test/test_vmap.py

```diff
@@ -733,6 +733,25 @@ def test_nn_module(self):
         result = vmap(model)(tensor)
         self.assertEqual(result, model(tensor))
 
+    def test_fallback_with_undefined_grad(self):
+        B0 = 7
+        x = torch.randn(2, 3, 4, 5, requires_grad=True)
+        weight = torch.randn(3, 3, 1, 1)
+        v = torch.randn(B0, 2, 3, 4, 5)
+
+        def get_vjp(v):
+            result = torch.nn.functional.conv2d(x, weight)
+            grad_x, = torch.autograd.grad(result, x, v)
+            return grad_x
+
+        # Runs vmap(get_vjp)(v), which should not error out.
```

Comment:
Note that a simple way to generate undefined gradients is with `out = torch._C._functions.UndefinedGrad()(inp)` which is going to ignore the gradient flowing for out and return and undefined gradient for inp.
Or with a custom autograd.Function that returns `None` for a given input (you might actually be able to use a custom Function to test your error message easily actually as `None` are propagated as undefined Tensors properly now).

Reply:
vmap doesn't see a custom autograd.Function as an operator, so the vmap fallback won't run on the autograd.Function (it'll run on torch operations inside the Function). A better way to test this would be to create a dummy operator that has a derivative formula that returns undefined tensors

Reply:
Ho right you are not at the autograd level but at operator. Sounds good.